pkg/kubelet/util/pluginwatcher/README

```diff
@@ -0,0 +1,29 @@
+This folder contains a utility, pluginwatcher, for Kubelet to register
+different types of node-level plugins such as device plugins or CSI plugins.
+It discovers plugins by monitoring inotify events under the directory returned by
+kubelet.getPluginsDir(). Lets refer this directory as PluginsSockDir.
+For any discovered plugin, pluginwatcher issues Registration.GetInfo grpc call
+to get plugin type, name and supported service API versions. For any registered plugin type,
+pluginwatcher calls the registered callback function with the received plugin
+name, supported service API versions, and the full socket path. The Kubelet
+component that receives this callback can acknowledge or reject the plugin
+according to its own logic, and use the socket path to establish its service
+communication with any API version supported by the plugin.
+
+Here are the general rules that Kubelet plugin developers should follow:
+- Run as priviledged. Currently creating socket under PluginsSockDir requires
+  priviledged access.
+- Implements the Registration service specified in
+  pkg/kubelet/apis/pluginregistration/v*/api.proto.
+- The plugin name sent during Registration.GetInfo grpc should be unique
+  for the given plugin type (CSIPlugin or DevicePlugin).
```

Comment:
Are there guidelines around namespacing? E.g. "mycompany.com/my-plugin"

Reply:
This is a list of plugin APIs that kubelet natively supports whose cardinality is very limited. The names of individual plugins on the other hand needs to adhere to existing namespacing norms.

Reply:
> Are there guidelines around namespacing? E.g. "mycompany.com/my-plugin"

Yes, it should be unique because socket name is expected to have one-to-one mapping with plugin name and it should not be hierarchical. Both points are mentioned.